cadcutils/cadcutils/net/ws.py

```diff
@@ -261,11 +242,11 @@ def _get_session(self):
         if self._session is None:
             self.logger.debug('Creating session.')
             self._session = RetrySession(self.retry)
-            if self.certificate_file_location is not None:
-                self._session.cert = (self.certificate_file_location, self.certificate_file_location)
+            if self.subject.certificate is not None:
+                self._session.cert = (self.subject.certificate, self.subject.certificate)
             else:
-                if self.basic_auth is not None:
-                    self._session.auth = self.basic_auth
+                if self.subject.get_auth(self.host) is not None:
+                    self._session.auth = self.subject.get_auth(self.host)
 
         user_agent = "{} {} {} {} ({})".format(self.agent, self.package_info, self.python_info,
                                                self.system_info, self.os_info)
```

Comment:
The above definitions of os_info don't have the '/' between the keyword/value pair which the other values here do, but perhaps that's OK as format is different.